storage/filesystem.go

```diff
@@ -47,6 +48,18 @@ type Filesystem struct {
 
 	// used by the 'read only' methods
 	fs fs.FS
+
+	// FixedFSGroup is an optional field which will set the gid ownership of all
+	// volume's data directories to this value.
+	// If this value is set, FSGroupVolumeAttributeKey has no effect.
+	FixedFSGroup *int64
+
+	// FSGroupVolumeAttributeKey is an optional well-known key in the volume
+	// attributes. If this attribute is present in the context when writing
+	// files, gid ownership of the volume's data directory will be changed to
+	// the value. Attribute value must be a valid int64 value.
+	// If FixedFSGroup is defined, this field has no effect.
+	FSGroupVolumeAttributeKey *string
```

Comment:
I think we could make this a non-ptr and just check for it being empty? (as we don't need to disambiguate between the empty value and not set, unlike `FixedFSGroup`)

Reply:
(I only raise it because if this is a pointer, there is the possibility that someone could change it at runtime/midway through execution.. which is.. weird 😅)

Reply:
Fine by me